============================================================================================== 
Warning! Mixing Conda and module environments may lead to corruption of the
user environment. 
We do not recommend users mixing those two environments unless absolutely
necessary. Note that 
SURF does not provide any support for Conda environment.
For more information, please refer to our software policy page:
https://servicedesk.surf.nl/wiki/display/WIKI/Software+policy+Snellius#SoftwarepolicySnellius-UseofAnacondaandMinicondaenvironmentsonSnellius 

Remember that many packages have already been installed on the system and can
be loaded using 
the 'module load <package__name>' command. If you are uncertain if a package is
already available 
on the system, please use 'module avail' or 'module spider' to search for it.
============================================================================================== 
[rank: 0] Seed set to 0
Using 16bit Automatic Mixed Precision (AMP)
GPU available: True (cuda), used: True
TPU available: False, using: 0 TPU cores
HPU available: False, using: 0 HPUs
LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [MIG-e4b2412b-d25c-54af-be2b-72f78b86af67]
Namespace(txt_file='./assets/book_EN_grimms_fairy_tales.txt', model_type='gpt-mini', block_size=128, use_pretrained=False, abs_emb=False, train_batch_size=128, generate_batch_size=5, generate_every_n_steps=1000, learning_rate=0.0005, weight_decay=0.1, betas=(0.9, 0.95), num_epochs=5, clip_grad_norm=1.0, log_dir='./logs', seed=0, num_workers=8, progress_bar=False, use_flash_attn=False, precision='16-mixed', compile=True, pretrained_tokenizer=False, device='cuda')
data has 540241 characters, 87 unique.
True False
number of parameters: 10.73M
running on device cpu
Finding best initial lr:   0%|          | 0/100 [00:00<?, ?it/s]Traceback (most recent call last):
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/torch/_dynamo/output_graph.py", line 1446, in _call_user_compiler
    compiled_fn = compiler_fn(gm, self.example_inputs())
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/torch/_dynamo/repro/after_dynamo.py", line 129, in __call__
    compiled_gm = compiler_fn(gm, example_inputs)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/torch/__init__.py", line 2235, in __call__
    return compile_fx(model_, inputs_, config_patches=self.config)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/torch/_inductor/compile_fx.py", line 1521, in compile_fx
    return aot_autograd(
           ^^^^^^^^^^^^^
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/torch/_dynamo/backends/common.py", line 72, in __call__
    cg = aot_module_simplified(gm, example_inputs, **self.kwargs)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/torch/_functorch/aot_autograd.py", line 1071, in aot_module_simplified
    compiled_fn = dispatch_and_compile()
                  ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/torch/_functorch/aot_autograd.py", line 1056, in dispatch_and_compile
    compiled_fn, _ = create_aot_dispatcher_function(
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/torch/_functorch/aot_autograd.py", line 522, in create_aot_dispatcher_function
    return _create_aot_dispatcher_function(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/torch/_functorch/aot_autograd.py", line 759, in _create_aot_dispatcher_function
    compiled_fn, fw_metadata = compiler_fn(
                               ^^^^^^^^^^^^
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/torch/_functorch/_aot_autograd/jit_compile_runtime_wrappers.py", line 382, in aot_dispatch_autograd
    fw_module, bw_module = aot_config.partition_fn(
                           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/torch/_inductor/compile_fx.py", line 1448, in partition_fn
    _recursive_joint_graph_passes(graph)
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/torch/_inductor/compile_fx.py", line 281, in _recursive_joint_graph_passes
    joint_graph_passes(gm)
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/torch/_inductor/fx_passes/joint_graph.py", line 456, in joint_graph_passes
    constant_fold_uniform_value(graph)
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/torch/_inductor/fx_passes/joint_graph.py", line 333, in constant_fold_uniform_value
    cf.run()
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/torch/_inductor/constant_folding.py", line 228, in run
    return super().run(initial_env=env)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/torch/fx/interpreter.py", line 146, in run
    self.env[node] = self.run_node(node)
                     ^^^^^^^^^^^^^^^^^^^
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/torch/_inductor/constant_folding.py", line 187, in run_node
    out = self._deduce_value(node)
          ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/torch/_inductor/fx_passes/joint_graph.py", line 320, in _deduce_value
    return node.target(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/torch/_ops.py", line 716, in __call__
    return self._op(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
RuntimeError: value cannot be converted to type at::Half without overflow

While executing %scalar_tensor : [num_users=1] = call_function[target=torch.ops.aten.scalar_tensor.default](args = (-1000000000.0,), kwargs = {dtype: torch.float16, layout: torch.strided, device: cuda:0})
Original traceback:
  File "/gpfs/home6/scur2773/uvadlc_practicals_2024/assignment2/part2/gpt.py", line 443, in forward
    x = block(x)    # Shape remains (b, t, n_embd)
  File "/gpfs/home6/scur2773/uvadlc_practicals_2024/assignment2/part2/gpt.py", line 222, in forward
    out = x + self.attn(self.ln1(x))  # Shape: (B, T, C)
  File "/gpfs/home6/scur2773/uvadlc_practicals_2024/assignment2/part2/gpt.py", line 169, in forward
    att = att.masked_fill(self.mask[:, :, :T, :T] == 0, -1e9)


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/gpfs/home6/scur2773/uvadlc_practicals_2024/assignment2/part2/train.py", line 209, in <module>
    train(args=args)
  File "/gpfs/home6/scur2773/uvadlc_practicals_2024/assignment2/part2/train.py", line 202, in train
    trainer.fit(lightning_model)
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/pytorch_lightning/trainer/trainer.py", line 538, in fit
    call._call_and_handle_interrupt(
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/pytorch_lightning/trainer/call.py", line 47, in _call_and_handle_interrupt
    return trainer_fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/pytorch_lightning/trainer/trainer.py", line 574, in _fit_impl
    self._run(model, ckpt_path=ckpt_path)
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/pytorch_lightning/trainer/trainer.py", line 961, in _run
    call._call_callback_hooks(self, "on_fit_start")
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/pytorch_lightning/trainer/call.py", line 218, in _call_callback_hooks
    fn(trainer, trainer.lightning_module, *args, **kwargs)
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/pytorch_lightning/callbacks/lr_finder.py", line 130, in on_fit_start
    self.lr_find(trainer, pl_module)
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/pytorch_lightning/callbacks/lr_finder.py", line 113, in lr_find
    self.optimal_lr = _lr_find(
                      ^^^^^^^^^
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/pytorch_lightning/tuner/lr_finder.py", line 278, in _lr_find
    _try_loop_run(trainer, params)
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/pytorch_lightning/tuner/lr_finder.py", line 523, in _try_loop_run
    loop.run()
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/pytorch_lightning/loops/fit_loop.py", line 205, in run
    self.advance()
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/pytorch_lightning/loops/fit_loop.py", line 363, in advance
    self.epoch_loop.run(self._data_fetcher)
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/pytorch_lightning/loops/training_epoch_loop.py", line 140, in run
    self.advance(data_fetcher)
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/pytorch_lightning/loops/training_epoch_loop.py", line 250, in advance
    batch_output = self.automatic_optimization.run(trainer.optimizers[0], batch_idx, kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/pytorch_lightning/loops/optimization/automatic.py", line 190, in run
    self._optimizer_step(batch_idx, closure)
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/pytorch_lightning/loops/optimization/automatic.py", line 268, in _optimizer_step
    call._call_lightning_module_hook(
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/pytorch_lightning/trainer/call.py", line 167, in _call_lightning_module_hook
    output = fn(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/pytorch_lightning/core/module.py", line 1306, in optimizer_step
    optimizer.step(closure=optimizer_closure)
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/pytorch_lightning/core/optimizer.py", line 153, in step
    step_output = self._strategy.optimizer_step(self._optimizer, closure, **kwargs)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/pytorch_lightning/strategies/strategy.py", line 238, in optimizer_step
    return self.precision_plugin.optimizer_step(optimizer, model=model, closure=closure, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/pytorch_lightning/plugins/precision/amp.py", line 78, in optimizer_step
    closure_result = closure()
                     ^^^^^^^^^
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/pytorch_lightning/loops/optimization/automatic.py", line 144, in __call__
    self._result = self.closure(*args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/torch/utils/_contextlib.py", line 116, in decorate_context
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/pytorch_lightning/loops/optimization/automatic.py", line 129, in closure
    step_output = self._step_fn()
                  ^^^^^^^^^^^^^^^
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/pytorch_lightning/loops/optimization/automatic.py", line 317, in _training_step
    training_step_output = call._call_strategy_hook(trainer, "training_step", *kwargs.values())
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/pytorch_lightning/trainer/call.py", line 319, in _call_strategy_hook
    output = fn(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/pytorch_lightning/strategies/strategy.py", line 390, in training_step
    return self.lightning_module.training_step(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/gpfs/home6/scur2773/uvadlc_practicals_2024/assignment2/part2/train.py", line 43, in training_step
    logits = self.forward(x)
             ^^^^^^^^^^^^^^^
  File "/gpfs/home6/scur2773/uvadlc_practicals_2024/assignment2/part2/train.py", line 39, in forward
    return self.model(x)
           ^^^^^^^^^^^^^
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/torch/_dynamo/eval_frame.py", line 465, in _fn
    return fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/torch/_dynamo/convert_frame.py", line 1269, in __call__
    return self._torchdynamo_orig_callable(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/torch/_dynamo/convert_frame.py", line 1064, in __call__
    result = self._inner_convert(
             ^^^^^^^^^^^^^^^^^^^^
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/torch/_dynamo/convert_frame.py", line 526, in __call__
    return _compile(
           ^^^^^^^^^
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/torch/_dynamo/convert_frame.py", line 924, in _compile
    guarded_code = compile_inner(code, one_graph, hooks, transform)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/torch/_dynamo/convert_frame.py", line 666, in compile_inner
    return _compile_inner(code, one_graph, hooks, transform)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/torch/_utils_internal.py", line 87, in wrapper_function
    return function(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/torch/_dynamo/convert_frame.py", line 699, in _compile_inner
    out_code = transform_code_object(code, transform)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/torch/_dynamo/bytecode_transformation.py", line 1322, in transform_code_object
    transformations(instructions, code_options)
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/torch/_dynamo/convert_frame.py", line 219, in _fn
    return fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/torch/_dynamo/convert_frame.py", line 634, in transform
    tracer.run()
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/torch/_dynamo/symbolic_convert.py", line 2796, in run
    super().run()
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/torch/_dynamo/symbolic_convert.py", line 983, in run
    while self.step():
          ^^^^^^^^^^^
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/torch/_dynamo/symbolic_convert.py", line 895, in step
    self.dispatch_table[inst.opcode](self, inst)
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/torch/_dynamo/symbolic_convert.py", line 2987, in RETURN_VALUE
    self._return(inst)
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/torch/_dynamo/symbolic_convert.py", line 2972, in _return
    self.output.compile_subgraph(
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/torch/_dynamo/output_graph.py", line 1142, in compile_subgraph
    self.compile_and_call_fx_graph(tx, pass2.graph_output_vars(), root)
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/torch/_dynamo/output_graph.py", line 1369, in compile_and_call_fx_graph
    compiled_fn = self.call_user_compiler(gm)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/torch/_dynamo/output_graph.py", line 1416, in call_user_compiler
    return self._call_user_compiler(gm)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/scur2773/.conda/envs/dl2024/lib/python3.12/site-packages/torch/_dynamo/output_graph.py", line 1465, in _call_user_compiler
    raise BackendCompilerFailed(self.compiler_fn, e) from e
torch._dynamo.exc.BackendCompilerFailed: backend='inductor' raised:
RuntimeError: value cannot be converted to type at::Half without overflow

While executing %scalar_tensor : [num_users=1] = call_function[target=torch.ops.aten.scalar_tensor.default](args = (-1000000000.0,), kwargs = {dtype: torch.float16, layout: torch.strided, device: cuda:0})
Original traceback:
  File "/gpfs/home6/scur2773/uvadlc_practicals_2024/assignment2/part2/gpt.py", line 443, in forward
    x = block(x)    # Shape remains (b, t, n_embd)
  File "/gpfs/home6/scur2773/uvadlc_practicals_2024/assignment2/part2/gpt.py", line 222, in forward
    out = x + self.attn(self.ln1(x))  # Shape: (B, T, C)
  File "/gpfs/home6/scur2773/uvadlc_practicals_2024/assignment2/part2/gpt.py", line 169, in forward
    att = att.masked_fill(self.mask[:, :, :T, :T] == 0, -1e9)


Set TORCH_LOGS="+dynamo" and TORCHDYNAMO_VERBOSE=1 for more information


You can suppress this exception and fall back to eager by setting:
    import torch._dynamo
    torch._dynamo.config.suppress_errors = True

Finding best initial lr:   0%|          | 0/100 [00:04<?, ?it/s]
srun: error: gcn2: task 0: Exited with exit code 1
srun: Terminating StepId=8690009.0

JOB STATISTICS
==============
Job ID: 8690009
Cluster: snellius
User/Group: scur2773/scur2773
State: FAILED (exit code 1)
Nodes: 1
Cores per node: 9
CPU Utilized: 00:00:18
CPU Efficiency: 6.90% of 00:04:21 core-walltime
Job Wall-clock time: 00:00:29
Memory Utilized: 2.01 MB
Memory Efficiency: 0.00% of 60.00 GB
